### ISSUE TITLE
Fix exception raised for get_file in client.py & Performance improvement

### ISSUE THREAD

**Issue by uhasa14** [pyrogram/pyrogram#1330]
### Checklist

- [X] I believe the idea is awesome and would benefit the framework
- [X] I have searched in the issue tracker for similar requests, including closed ones

### Description

- When using `download_media` using userbot, if any exception occurs like `400 AUTH_BYTES_INVALID` or `420 FLOOD_WAIT_X` in `get_file` function during **importing or exporting authorization**; the **exception gets registered in logs but do not get cached by any exception handler**. 

- **Store and Reuse media sessions** so that FloodWait is not raised for repeated authorization requests when downloading multiple files from DC other than self one. *(Like it used to do before this commit https://github.com/pyrogram/pyrogram/commit/8441ce2f4790f81655fe48f6fda57dad432ee798)*

Apologies if I did any mistake. Writing an issue for the first time.
Thanks.

**Comment by uhasa14** [comment]
https://github.com/pyrogram/pyrogram/blob/efac17198b5fcaec1c2628c4bba0c288a4d617d4/pyrogram/client.py#L1014-L1023
- When trying to raise any exception in *Line 1015 & 1019*, we get the following error in terminal:
`expected str, bytes or os.PathLike object, not NoneType`
`stat: path should be string, bytes, os.PathLike or integer, not NoneType`

PS: Happens when using `download_media` method.

**Comment by KurimuzonAkuma** [comment]
> * **Store and Reuse media sessions** so that FloodWait is not raised for repeated authorization requests when downloading multiple files from DC other than self one. _(Like it used to do before this commit [8441ce2](https://github.com/pyrogram/pyrogram/commit/8441ce2f4790f81655fe48f6fda57dad432ee798))_

#1313 works fine

**Comment by uhasa14** [comment]
> #1313 works fine

Thanks for this. Do you have solution to other problems too?

**Comment by KurimuzonAkuma** [comment]
> > #1313 works fine
> 
> Thanks for this. Do you have solution to other problems too?

No, just this 🥲

**Comment by gjp1120** [comment]
also hit the first problem, when this happens `message.download()` will generate 0 byte files, no exception raised

**Comment by uhasa14** [comment]
> also hit the first problem, when this happens `message.download()` will generate 0 byte files, no exception raised

Yup that's annoying sometimes. Did you find any workaround for this?

**Comment by abcdenis** [comment]
> > also hit the first problem, when this happens `message.download()` will generate 0 byte files, no exception raised
> 
> Yup that's annoying sometimes. Did you find any workaround for this?


Hi,
there are an explanation and workaround in this thread:
https://github.com/pyrogram/pyrogram/issues/1306

**Comment by uhasa14** [comment]
> It seems that we have a fix for this awaiting to be accepted into the main branch with #1348 :)
Seems that pyrogram is no longer maintained. Do you know any fork which is being updated?

**Comment by KurimuzonAkuma** [comment]
> > It seems that we have a fix for this awaiting to be accepted into the main branch with #1348 :)
> 
> Seems that pyrogram is no longer maintained. Do you know any fork which is being updated?

https://github.com/KurimuzonAkuma/pyrogram/tree/dev

**Comment by abcdenis** [comment]
Another dirty fix: if there is disk logging with mesages from pyrogram then its easy to write class that will be search FloodWait in log file since last call and extract sleep time from it :-/
